lint_test/avoid_using_api/named_parameter_ban/lib/named_parameter_ban_test.dart

```diff
@@ -0,0 +1,28 @@
+// ignore_for_file: unused_local_variable
+
+import 'package:external_source/external_source.dart';
+
+void testingBannedCodeLint() async {
+  // expect_lint: avoid_using_api
+  final bannedCodeUsage = BannedCodeUsage(parameter1: '');
+
+  // expect_lint: avoid_using_api
+  final test3 = BannedCodeUsage.test3(parameter1: '');
+
+  // expect_lint: avoid_using_api
+  final test5 = BannedCodeUsage.test5(parameter1: '');
+
+  // expect_lint: avoid_using_api
+  BannedCodeUsage.test2(
+    parameter1: 'test',
+  );
+
+  final obj = BannedCodeUsage();
+  // expect_lint: avoid_using_api
+  obj.test(parameter1: '');
+
+  test(parameter1: '');
+
+  // expect_lint: avoid_using_api
+  2.banned(parameter1: '');
```

Comment:
It would be great to add cases when we do not expect to have a lint for each test case, so we can be sure it does not generate false positives.